aten/src/ATen/native/SharedReduceOps.h

```diff
@@ -210,17 +210,17 @@ struct NormOps {
   }
 };
 
-template <typename acc_t>
+template <typename acc_t, typename data_t, typename out_t>
 struct NormZeroOps {
-  inline C10_DEVICE acc_t reduce(acc_t acc, acc_t data, int64_t /*idx*/) const {
-    return acc + (data==acc_t(0) ? acc_t(0) : acc_t(1));
+  inline C10_DEVICE acc_t reduce(acc_t acc, data_t data, int64_t /*idx*/) const {
+    return acc + (data==data_t(0) ? acc_t(0) : acc_t(1));
   }
 
   inline C10_DEVICE acc_t combine(acc_t a, acc_t b) const {
     return a + b;
   }
 
-  inline C10_DEVICE acc_t project(acc_t a) const {
```

Comment:
`a` is being passed as `acc_t` type and will be returned as `out_t` type without any casting. will that work?

Reply:
Thanks your comments. Fixed it.